product-service/productById.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+const products = require("./products");
+
+module.exports.getProductById = async (event) => {
+  try {
+    let id = null; 
+    if (event.pathParameters) {
+      id =  event.pathParameters.id;
+    }
+    if (!id)
+      return {
+        headers: {
+          "Access-Control-Allow-Origin": "*",
```

Comment:
According to DRY principle would be nice to store headers in a separate variable as long as they repeat several times.